src/org/labkey/test/components/bootstrap/LoggedOutDialog.java

```diff
@@ -0,0 +1,118 @@
+{
+    static public List<Locator> LKS_TOP_LEVEL_ELEMENTS = List.of(org.labkey.test.Locators.headerContainer(),
+            org.labkey.test.Locators.bodyPanel(), org.labkey.test.Locators.footerPanel());
+    static public List<Locator> APP_TOP_LEVEL_ELEMENTS = List.of(org.labkey.test.Locators.bodyPanel());
+
+    protected LoggedOutDialog(WebDriver driver)
+    {
+        super(new ModalDialogFinder(driver).withTitle("Logged Out")
+                .waitFor().getComponentElement(), driver);
+    }
+
+    static public LoggedOutDialog waitFor(WebDriver driver)
+    {
+        return new LoggedOutDialog(driver);
+    }
+
+    public void clickReloadPage()
+    {
+        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().reloadPageBtn));
+        elementCache().reloadPageBtn.click();
+    }
+
+    public void reloadPageAndConfirmPostLoginRedirect(String expectedURL)
+    {
+        clickReloadPage();  // dismisses the dialog
+
+        // at this point, we're either at a login page already, or at a public page showing a link to one
+        // when we add biologics to this, there will likely be a third case to handle here
+        WebElement el = Locator.waitForAnyElement(new FluentWait<SearchContext>(getDriver()).withTimeout(Duration.ofMillis(WAIT_FOR_JAVASCRIPT)),
+                org.labkey.test.Locators.signInLink,
+                Locator.tagWithName("form", "login"));
+
+        if (el.getAttribute("class").equals("header-link"))
+            getWrapper().clickAndWait(el);
+
+        // We should now be on the regular login page, without the beginAt action simpleSignIn does
+        getWrapper().assertElementPresent(Locator.tagWithName("form", "login"));
+        getWrapper().setFormElement(Locator.name("email"), PasswordUtil.getUsername());
+        getWrapper().setFormElement(Locator.name("password"), PasswordUtil.getPassword());
+        getWrapper().clickButton("Sign In", 0);
+
+        var returnUrl = getWrapper().getCurrentRelativeURL();
+        assertThat("Signing back in did not take us back to the expected page.", returnUrl, startsWith(expectedURL));
+    }
+
+    /**
+     *  Checks the background/masked elements behind the dialog to ensure that the expected content
+     *  areas are blurred/styled as expected.
+     *  If any of them are not present or not styled as expected, will return false.
+     * @return True if all page sections that are present are styled with 'lk-content-blur', otherwise false
+     */
+    public boolean isContentBlurred(List<Locator> elementsToBeBlurred)
+    {
+        // if any of these bodyParts are present and don't have the lk-content-blur style applied, return false
+        // otherwise, true
+        for(Locator loc : elementsToBeBlurred)
+        {
+            if(!loc.findElement(getDriver()).getAttribute("class").contains("lk-content-blur"))
+                return false;
+        }
+        return true;
+    }
```

Comment:
Since we check the blurred an unblurred conditions, I suggest we check all of the elements and have it throw an error if there is a mix of blurriness.